en/lessons/basics/operators.md

```diff
@@ -0,0 +1,185 @@
+---
+version: 1.0.0
+title: Operators
+---
+
+Haskell provides constructs for using infix binary functions, called operators. In
+this lesson you will learn how to define operators and control the parsing of
+operators, and how infix and prefix syntax can be interchanged.
+
+{% include toc.html %}
+
+
+## Using Operators
+
+Operators in Haskell are functions that can be placed between two arguments: they can
+return a value of any type and their names are either a string of symbols or ordinary
+function names surrounded in backticks. Using infix notation makes it easier to write
+expressions containing many common functions:
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> abs . subtract 119 $ 2 + 2
+115
+ghci> (`mod` 930233356) . negate $ 4
+930233352
+ghci> - (7 + 57843)
+-57850
+```
+__Note__: negate (`-`) is the only unary operator in Haskell.
+
+You may have seen operators such as `+`, `*`, and `.` before. The `$` operator is
+useful for removing parentheses in function application: `abs . subtract 1 $ 2 + 2` is
+equivalent to `(abs . subtract 1) (2 + 2)`. You can find more examples of operators on
+[FPComplete's Operator Glossary](https://www.fpcomplete.com/haskell/tutorial/operators/).
+
+### Interchanging Operators and Functions
+
+To use an operator symbol as if it were a regular function, surround it in parentheses.
+Furthermore, you can create partially applied operators, called sections, by putting
+an expression on only one side of the operator:
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> (+) 2 3
+5
+ghci> (5 <=) . (* 2) $ 3
+True
+```
+
+The parenthesised operators in the above example can be treated as equivalent to the following
+anonymous functions:
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> (\a b -> a + b) 2 3
+5
+ghci> (\a -> 5 <= a) . (\a -> a * 2) $ 8
+True
+```
+
+Conversely, to make a regular function identifier an operator, surround it in backticks:
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> 2010 `mod` 1998
+12
+ghci> (`div` 3) . (* 1471) $ 5
+2451
+```
+
+
+### Operator Precedence and Associativity
+
+An operator's precedence and associativity combined is called its fixity, and it
+controls how the operator is parsed in an expression. To find out the fixity of an
+operator, look through `ghci`'s `:info` or the documentation for the operator:
+if fixity information isn't displayed, it defaults to `infixl 9`. The same applies to
+regular functions used in infix mode.
```

Comment:
Can you explain what you mean by associativity? i.e. that the prefix of `r` or `l` in the `infixl 9` declaration indicates the direction in which it associates. A really good illustration of this is the `esqueleto` experimental syntax operator `:&`  which allows you to pattern match and ignore all arguments to the left because of its associativity:

```haskell
(a :& b :& c :& d :& e) == (_ :& d :& e)
```

Reply:
I cover a process that explains how associativity works and then provide a couple examples later on. Do you think these examples build enough intuition? 

I feel like the most intuitive examples of associativity are in data constructors, such as that for list, but I'm not sure I should mention that here.

Reply:
Yeah, this subject does get more advanced very quickly if you delve into to much detail. Again, I leave it to your discretion.